### PR TITLE
EZPZ_CU-369rrr7: Create Reset Password Page

### PR DESCRIPTION
**Pull request by KevinLaCarrubba**
- Created a Reset Password page
- Added on submit validations to check if both password match
- Created an error field area to display the message 
*****
***Side Task***
- Fixed an overlooked classname issue in forgot password field. 
- Added in a message prop for Password field which is not required. It takes in a boolean type and will control wether or not the password criteria message will be displayed.

### CODE REVIEW

**Comment by arfusop** [comment]
Task linked: [CU-369rrr7 Create the Reset Password page](https://app.clickup.com/t/369rrr7)

File: pages/password/reset/index.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import ResetPassword from '../../../components/views/auth/ResetPassword'
```
Comment (arfusop) [inline pages/password/reset/index.tsx:1]: Not sure why you deleted my page with the token. Remove this page, move this code into that page after it was restored. We need the token to be passed in via the page url for authentication purposes on both the UI and server

File: components/form/fields/PasswordField/index.tsx [hunk 2]
```diff
@@ -46,6 +48,13 @@ const PasswordField = ({
                 htmlFor={id}>
                 {placeholder}
             </label>
+            {message && (
```
Comment (arfusop) [inline components/form/fields/PasswordField/index.tsx:51]: Build this out to include the message if message is provided

File: components/views/auth/ResetPassword.tsx [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+import { BiErrorAlt } from 'react-icons/bi'
+
+import styles from './AuthPages.module.scss'
+
+const INITIAL_STATE = {
+    newPassword: { value: '', error: '' },
+    confirmPassword: { value: '', error: '' }
+} as FormProps
+
+const PASSWORD_ERROR_MESSAGE = 'Password does not meet requirements'
```
Comment (arfusop) [inline components/views/auth/ResetPassword.tsx:15]: This needs to be specific to the requirements of the password instead

File: components/views/auth/ResetPassword.tsx [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+const VALIDATIONS = {
+    newPassword: {
+        test: (value: string) => VALID_PASSWORD.test(value),
+        message: PASSWORD_ERROR_MESSAGE
+    },
+    confirmPassword: {
+        test: (value: string) => VALID_PASSWORD.test(value),
+        message: PASSWORD_ERROR_MESSAGE
```
Comment (arfusop) [inline components/views/auth/ResetPassword.tsx:23]: This should be a message like `Both passwords must match`

File: components/views/auth/ResetPassword.tsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+        if (
+            confirmPassword.value !== '' &&
+            newPassword.value !== confirmPassword.value
+        ) {
+            setSubmitError(true)
+        } else {
+            setSubmitError(false)
+        }
+    }
+    useEffect(() => {
```
Comment (arfusop) [inline components/views/auth/ResetPassword.tsx:48]: insert line break 😉

File: components/views/auth/ResetPassword.tsx [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        Object.keys(form).forEach(item => {
+            const current = form[item]
+            if ((isFormValid && !current.value) || current.error) {
+                isFormValid = false
+            }
+        })
+
+        setDisableSubmit(!isFormValid)
+    })
+    return (
```
Comment (arfusop) [inline components/views/auth/ResetPassword.tsx:59]: line break

File: components/views/auth/ResetPassword.tsx [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+            <div className={styles.formContainer}>
+                <h1 className={styles.header}>Reset Password</h1>
+                <form onSubmit={handleSubmit}>
+                    {submitError && (
+                        <div className={styles.errorField}>
+                            <BiErrorAlt className={styles.errorIcon} />
+                            The passwords do not match
+                        </div>
+                    )}
+
```
Comment (arfusop) [inline components/views/auth/ResetPassword.tsx:70]: no need for line break 🤣

**Review by arfusop — CHANGES_REQUESTED** [review]

File: components/views/auth/ResetPassword.tsx (hunk 3 above)
Comment (KevinLaCarrubba) [reply components/views/auth/ResetPassword.tsx:15]: done

**Review by arfusop — APPROVED** [review]

**Review by SamiFdev — APPROVED** [review]

**Pull request merged**